lib/workers/repository/updates/branch-name.ts

```diff
@@ -42,16 +43,36 @@ export function generateBranchName(update: RenovateConfig): void {
       update.groupSlug = `patch-${update.groupSlug}`;
     }
     update.branchTopic = update.group.branchTopic || update.branchTopic;
-    update.branchName = template.compile(
-      update.group.branchName || update.branchName,
-      update
-    );
+    update.branchName = update.group.branchName || update.branchName;
+  }
+
+  if (update.maxBranchLength) {
+    let hashLength = update.maxBranchLength - update.branchPrefix.length;
+    if (hashLength <= 0) {
```

Comment:
```suggestion
    if (hashLength <= 10) {
```

If we default to 10, let's make that the minimum. Although maybe less is OK?